RAINCATCH-1118 - introduction section

austincunningham opened the pull request:
**Motivation :**
need an introduction section in documentation

**What**
Added a overview section to the docs

- [x] Added overview section

**Additional info**
https://issues.jboss.org/browse/RAINCATCH-1118

austincunningham committed:
added overview documentation

austincunningham committed:
Added overview docs

wtrocki commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+# Created by .ignore support plugin (hsz.mobi)
```
Remove this :)

wtrocki commented on docs/workforce-management-framework/topics/con-overview.adoc:
```diff
@@ -0,0 +1,21 @@
+To quickly start write mobile enabled Workflow Management applications please follow our guides:
+
+- link:./guides/wfm.html[WFM Guide]
+Getting Started for Workflow Management Solutions.
```
Do we still support this? I think that actually modular documentation it's one user oriented guide :)

austincunningham replied:
out of date

wtrocki commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+== Introduction
+
+{WFM-RC-NameLong}  consist of set of node.js modules written in TypeScript language that are packaged and distributed as
```
That statement is not fully true. replace `node.js` with `npm modules` - Node.js suggest that our modules are server side while majority of them are running in the browser.

austincunningham replied:
agree

wtrocki commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+Solution consist of two fully independent frameworks:
+
+ * Workflow Management Framework
+ * Rapid Application Development Framework
```
That's no longer valid. It's WFM and Security.

austincunningham replied:
agree

wtrocki commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+
+===  Workflow Management Framework
+
+{WFM-RC-NameLong} Workflow Management Framework build BPMN compliant workflow management solution.
```
We aren't `BPMN compliant` etc.  Wort researching how wfm is being described and add it here.

austincunningham replied:
agree

jstaffor committed:
Update con-overview.adoc

Reduced this module to just concept information. Reference material removed will be added into the ref module I will create as part of this commit.

austincunningham committed:
merged master

austincunningham committed:
Merge branch 'RAINCATCH-1118' of github.com:feedhenry-raincatcher/raincatcher-docs into RAINCATCH-1118

wtrocki commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+===  Workflow Management Framework
+
+{WFM-RC-NameLong} Workflow Management Framework build BPMN compliant workflow management solution.
+WFM framework allows to build advanced Business Process Management solutions.
```
This stuff was written by myself but it;s a little bit vague. I think it's best to research what WFM solutions provide and use some business related words here.

austincunningham replied:
same as above

wtrocki commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+WFM framework allows to build advanced Business Process Management solutions.
+Framework helps to write solutions that connects a business back-office with its fleet of mobilized employees.
+
+ * Task: defines single unit of work for business process
```
ping @paolobueno - I think we going to move this back to the original names. Can you confirm?

jstaffor committed:
Rename con-overview.adoc to con-raincatcher-overview.adoc

Updated heading to match my previous commit

jstaffor committed:
Update overview-of-raincatcher.adoc

Sync'd file name with updated in this commit

wtrocki commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+ * Process: instance of flow containing current progress and business objects.
+
+Users can write custom Tasks (Workflow Steps) that will implement any business logic and link implementation with any UI solution.
+Framework abstract from any web or mobile application libraries, allowing developers to focus on implementation of the process.
```
No longer actual. We need to provide here a more context about technology stack.

austincunningham replied:
removed outdated and covered in next section

wtrocki commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+Users can write custom Tasks (Workflow Steps) that will implement any business logic and link implementation with any UI solution.
+Framework abstract from any web or mobile application libraries, allowing developers to focus on implementation of the process.
+
+===  Rapid Application Development Framework
```
That's no longer actual.

austincunningham replied:
removed no longer actual

wtrocki commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+
+image:../../shared/images/technologies.png[Technologies]
+
+Main targets for framework:
```
Targets are still valid but we need to reformat them properly and add some more context that was build over the time.

austincunningham replied:
changed from targets to goals

wtrocki commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+
+===  Demo application
+
+- Set of example implementations for Workflow Management Solutions (Workorder, WorkorderResult)
```
Let's mention that we have `Demo purpose examples etc.` not production ready. Just to showcase modules.

austincunningham replied:
done

wtrocki commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+|===
+| vehicle-inspection-task
+
+|interface VehicleAssessment extends Task
```
No longer actual.

austincunningham replied:
changed

wtrocki commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+
+===  Module structure
+
+RainCatcher modules are written in link:http://typescriptlang.org[TypeScript].
```
Not entirely. It's safe to say "core components"

austincunningham replied:
agree

jstaffor commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+== Introduction
```
Remove lines 1 --> 14 (motivation is to conceptualize main areas of this text (for eg: Workflow Management Framework)   into concept mod docs and add accompanying reference mod docs). We can import then into the User Story (and possibly reuse concepts/references in RC features section)

austincunningham replied:
was merged

jstaffor commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+
+As extension to frameworks above {ProductName} provides set of example module implementations as part of demo applications.
+
+===  Workflow Management Framework
```
Create a concept adoc called "con-workflow-management-framework.adoc" and add lines 15 --> 27 into it.
(It might make more sense to you to add more of less lines in (ie; up to line 50 for eg) - just make the call on this)

Locate it in the topics folder.

Import into the "overview-of-raincatcher.adoc" which you rightly created as the user story +1

I'll make edits to the text in this file then

austincunningham replied:
was merged

jstaffor commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+
+Framework abstract from any application logic and technologies allowing developers to extend it for their needs.
+
+===  Demo application
```
Create a concept adoc called "con-demo.adoc" and add lines <not sure> into it.
(We will DEFO end up add to this over time and reusing the many many time)

Locate it in the topics folder.

Import into the "con-demo.adoc" which you rightly created as the user story +1

I'll make edits to the text in this file then

austincunningham replied:
was merged

jstaffor commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+
+===  Module structure
+
+RainCatcher modules are written in link:http://typescriptlang.org[TypeScript].
```
I think we need to create two ModDocs for type script:

1)  a con adoc called "con-wfm-typescript.adoc"
 - contains the wfm/typscript general info - concept stuff (ie: RainCatcher modules are written in typescript. The main files are modules that by default export a class containing the main implementation for the module's intent,with a named export containing a public interface that should be depended upon, and re-implemented). Then in the text we have "for more inforation, see <ref-wfm-typescript.adoc>"

2)  a ref adoc called "ref-wfm-typescript.adoc"
 - contains "factual" info
  - eg: (RainCatcher modules are written in link:http://typescriptlang.org[TypeScript] and use version x.y. An example of RC typescript implementation is: 
```typescript
import Implementation, { PublicInterface } from '@raincatcher/module';
```
)

austincunningham replied:
was merged

wtrocki commented on docs/workforce-management-framework/topics/pro-overview-introduction.adoc:
```diff
@@ -0,0 +1,99 @@
+
+{WFM-RC-NameLong} Rapid Application Development Framework allows to build production-grade Node.js and Cordova based applications.
+
+image:../../shared/images/technologies.png[Technologies]
```
I'm not sure if we can use this image or anything that is not officially accepted and matches style guide. We may migrate that to text.

austincunningham replied:
removed , changed to table

jstaffor commented on docs/workforce-management-framework/upstream-1/overview-of-raincatcher.adoc:
```diff
@@ -0,0 +1,15 @@
+include::topics/shared/attributes.adoc[]
```
This file demonstrates to me that you have a handle on the repo, fair play Austin +1

austincunningham replied:
:)

austincunningham committed:
address pr comments

austincunningham committed:
Merge branch 'master' of github.com:feedhenry-raincatcher/raincatcher-docs into RAINCATCH-1118

wtrocki commented:
@austincunningham extension to this getting started will be the user story:
`As an user I want to see differences between Raincatcher Next and Raincatcher Beta`

I really see now how modular docs will work. We can reference to the same parts of the documentaions for different stores.

austincunningham committed:
merged structure changes

austincunningham committed:
merged structure changes

jstaffor committed:
RAINCATCH-1118v1 (#16)

* RAINCATCH-1118v1 - Wednesdays update

* RAINCATCH-1118v1 - Wednesdays update

* New anchor and xref logic added to negate build errors around this topic

* PRD-104.2 - Detailing Supported Modules

austincunningham committed:
fix render issues for asciidoc

austincunningham committed:
commented out modules that dont exist

wtrocki commented on README.adoc:
```diff
@@ -90,14 +90,31 @@ Build the example located in the _upstream-1_ folder:
 
 To best understand the information above, see the example located in the repo.
 
+==== Dynamic Anchors and Hard-Coded xref
```
Why this is here ? Rebase

jstaffor replied:
This week I encountered a new issue re the repo structure and xrefs. My solution is to hard code xref when faced with this situation. I wrote this up during the week and added it today.

It's in the readme because if other people faced the situation I encountered (which they will), I'll point them to this text.

wtrocki replied:
yeah.. i saw yor pr. I was asking why this change is in introduction PR :)

jstaffor replied:
Ahhhh-  it's there because I made this PR when _I didn't know how to properly work with upstream and forks_ - it's basically a headache I have caused (i'm guessing you are seeing conflicts as I made the change in (https://github.com/feedhenry-raincatcher/raincatcher-docs/pull/20) - my apologies .

austincunningham replied:
you change the base of john's fork to this branch .

jstaffor replied:
Once the very first few PRs are closed (they include my "dirty work"), this situation will become non existent as now that I know how the flow (upstream and forks) works

jstaffor replied:
Basically, what was pushed in (https://github.com/feedhenry-raincatcher/raincatcher-docs/pull/20) is the latest and greatest.

To save every bodies time right now is this:

1) copy all content from "README.adoc" in (https://github.com/feedhenry-raincatcher/raincatcher-docs/pull/20)  - it's been signed off by Eng and Docs, and subsequently merge 

2)and replace it with content in this PR readme.

Job done. Again, sorry for this situation, they will become will become non existent as now that I know how the flow (upstream and forks) works

austincunningham committed:
merged master

austincunningham committed:
remove todos and updated content

wtrocki commented:
Closing this PR as new one was created. 
https://github.com/feedhenry-raincatcher/raincatcher-docs/pull/21/files

Feel free to reopen or cherry pick some elements that did not land into master.
